lib/internal/repl.js

```diff
@@ -51,19 +54,31 @@ function createRepl(env, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY_FILE) {
-    return setupHistory(repl, env.NODE_REPL_HISTORY_FILE, cb);
+  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
+    return setupHistory(repl, env.NODE_REPL_HISTORY,
+                        env.NODE_REPL_HISTORY_FILE, cb);
   }
   repl._historyPrev = _replHistoryMessage;
   cb(null, repl);
 }
 
-function setupHistory(repl, historyPath, ready) {
-  const fs = require('fs');
+function setupHistory(repl, historyPath, oldHistoryPath, ready) {
+  if (!historyPath) {
+    try {
+      historyPath = path.join(os.homedir(), '.node_repl_history');
```

Comment:
OK then I see a need for doc improvement: https://iojs.org/api/os.html#os_os_homedir (unrelated to this PR, of course)